src/System.IO.Compression.Brotli/README.md

```diff
@@ -29,43 +29,41 @@ public static class Brotli {
+```State``` is used in both compress and decompress mode to save a temporary status and data of process.  State will be set automatically on initial call of Compress or Decompress. The state cannot be re-used for multiple data streams and cannot switch modes (compress vs. decompress).
 
-The ```Compress``` performs data compression. Taking data from ```source``` and writing compressed data to ```destination```.
+The ```Compress``` performs data compression. It takes data from ```source``` and writes compressed data to ```destination```.
 
 The ```FlushEncoder``` returns compressed data, which was sent to ```state``` using Compress. This method should always be called after all Compress executions (```isFinished = true```) or when you want to get compressed data immediately (```isFinished = false```).
 
-The ```Decompress``` Decompresses the data from ```source``` into ```destination```. 
+The ```Decompress``` decompresses the data from ```source``` into ```destination```. 
 
-```SetQuality``` allows you to set quality of compression ```0 to 11```. The higher quality means the higher compression ratio, but more compute time.
+```SetQuality``` allows you to set quality of compression ```between 0 to 11```. The higher quality means the higher compression ratio, but more compute time.
 
-```SetWindow``` - Logarithm of recommended sliding LZ77 window size. Encoder may reduce this value, e.g. if input is much smaller than window size. Window size is (1 << value) - 16. Possible values: ```11 to 24```
+```SetWindow``` - Logarithm of recommended sliding LZ77 window size. Encoder may reduce this value, e.g. if input is much smaller than window size. Window size is ```(1 << value) - 16```. Possible values: ```11 to 24```
 
 ### Examples
 Simple method to compress bytes to file.
 ```C#
 
- static void CompressSimple(byte[] bytes, string OutFile)
+ static void CompressSimple(byte[] bytes, string outFile)
  {
     Brotli.State state = new Brotli.State();
+    Span<byte> spanBytes = new Span<byte>(bytes);
     byte[] compressed = new byte[Brotli.GetMaximumCompressedSize(bytes.Length)];
-    state.SetQuality();
-    TransformationStatus result = Brotli.Compress(bytes, compressed, out int consumed, out int written, ref state);
+    TransformationStatus result = Brotli.Compress(spanBytes, compressed, out int consumed, out int written, ref state);
     while (result != TransformationStatus.Done)
     {
-        result = Brotli.Compress(bytes, compressed, out consumed, out written, ref state);
-        bytes = bytes.AsSpan().Slice(consumed).ToArray();
+        result = Brotli.Compress(spanBytes, compressed, out consumed, out written, ref state);
+        spanBytes = spanBytes.Slice(consumed);
```

Comment:
as I remember Slice return Span and don't change argument. So if we have multiply iterations of this loop we had or create a variable and do ```allConsumed += consumed``` and than can do ```spanBytes.Slice(allConsumed)``` or just update spanBytes every time.